beacon-chain/graffiti/graffiti-proposal-brief.md

```diff
@@ -0,0 +1,72 @@
+# Graffiti Version Info Implementation
+
+## Summary
+Add automatic EL+CL version info to block graffiti following [ethereum/execution-apis#517](https://github.com/ethereum/execution-apis/pull/517). Format: `GE168dPR63af` (Geth commit 168d + Prysm commit 63af). User graffiti always takes precedence.
+More details: https://github.com/ethereum/execution-apis/blob/main/src/engine/identification.md 
+
+## Implementation
+
+### Core Component: GraffitiInfo Struct
+Thread-safe struct holding version information:
+```go
+type GraffitiInfo struct {
+    mu           sync.RWMutex
+    userGraffiti string  // From --graffiti flag (set once at startup)
+    clCode       string  // "PR" (hardcoded)
+    clCommit     string  // From version.GetCommitPrefix() helper function
+    elCode       string  // From engine_getClientVersionV1
+    elCommit     string  // From engine_getClientVersionV1
+}
+```
+
+### Flow
+1. **Startup**: Parse flags, create GraffitiInfo with user graffiti and CL info. If user graffiti is set, log an informational message that custom graffiti overrides automatic client version reporting which helps track client diversity.
+2. **Wiring**: Pass struct to both execution service and RPC validator server
+3. **Runtime**: Execution service goroutine periodically calls `engine_getClientVersionV1` and updates EL fields
+4. **Block Proposal**: RPC validator server calls `GenerateGraffiti()` to get formatted graffiti
+
+### Priority Order
+```go
+func (g *GraffitiInfo) GenerateGraffiti() [32]byte {
+    if userGraffiti != "" {
```

Comment:
What do you think about implementing Mark's "[flexible standard](https://hackmd.io/@wmoBhF17RAOH2NZ5bNXJVg/BJX2c9gja)" so that we can use whatever space is leftover after the user's message to pack in the impl information?